src/ui/EditorNS/editor.cpp

```diff
@@ -383,9 +383,14 @@ namespace EditorNS
                 }
             }
         }
-        // If we make it here we need to make sure we're loaded.
-        waitAsyncLoad();
+        // We wait for the page to finish loading before doing this one.
         // First non-blank line
+        connect(this, &Editor::editorReady, this,
+                &Editor::detectLanguageFromContent);
```

Comment:
I'm still debating on how to do this cleanly... I might put a C_FUN_DETECT_LANGUAGE in the handlers to deduce the language instead.